Source/Concurrency/MoverCheck.cs

```diff
@@ -83,36 +83,53 @@ where first.LayerRange.OverlapsWith(second.LayerRange)
 
       foreach (var sequentialization in civlTypeChecker.Sequentializations)
```

Comment:
The body of this for loop is getting a bit unwieldy. Perhaps start by factoring out the body into another helper procedure. Then try to write your code. It could look more understandable.